akka-projection-core/src/main/scala/akka/projection/internal/TelemetrySpi.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.internal
+
+import scala.collection.immutable
+
+import akka.actor.typed.ActorSystem
+import akka.annotation.InternalApi
+import akka.annotation.InternalStableApi
+import akka.projection.ProjectionId
+
+/**
+ * INTERNAL API
+ */
+@InternalStableApi private[projection] object TelemetryProvider {
+  def start(projectionId: ProjectionId, system: ActorSystem[_]): Telemetry = {
+    val dynamicAccess = system.dynamicAccess
+    if (system.settings.config.hasPath("akka.projection.telemetry.fqcn")) {
+      val telemetryFqcn: String = system.settings.config.getString("akka.projection.telemetry.fqcn")
+      dynamicAccess
+        .createInstanceFor[Telemetry](
+          telemetryFqcn,
+          immutable.Seq((classOf[ProjectionId], projectionId), (classOf[ActorSystem[_]], system)))
+        .get
+    } else {
+      // TODO: review this default: should use a default value in reference.conf instead?
+      NoopTelemetry
+    }
+  }
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalStableApi private[akka] trait Telemetry {
```

Comment:
This whole file is out of place. It should not be in `..internal` nor be `private[akka]`.